timemachine/cpp/src/tibd_exchange_move.cu

```diff
@@ -208,7 +205,7 @@ void TIBDExchangeMove<RealType>::move(
         this->sampler_.sample_device(src_count, 1, d_src_weights_.data, this->d_samples_.data, stream);
 
         // Selected an index from the src weights, need to remap the samples idx to the mol indices
-        k_adjust_sample_idx<<<1, tpb, 0, stream>>>(
+        k_adjust_sample_idx<<<1, 1, 0, stream>>>(
```

Comment:
1 thread per block?

Reply:
This kernel just modifies one value. Doesn't need to run more than one thread.

Will modify the kernel to make this clearer, as else it risk being a race condition.

Reply:
Modified kernel to defend against this, though running one thread per block and only one block is still the desirable thing to do.

Reply:
nit/question: is there a reason to add `if (idx > 0) { return; }` to the kernel implementation in addition to this change? Seems like this could lead to confusion if we later increase the number of samples. (I didn't understand the comment about a possible race condition here)

Reply:
The old version simply would have multiple threads read from `sample_idxs[0]` and then use that value to override the same value. This is a potential race condition if one of the threads reads the value after it has been updated and updates it again. 

As to the confusion about supporting more number of samples, I think that confusion is interspersed through the code already. Some parts of the code early on made it seem like multiple samples was an easy introduction, but as I have gone along it has become clear that it wouldn't be straightforward. At the moment it seems best to leave to resolving this when we support multiple samples.